apps/remix-dapp/src/utils/txRunner.ts

```diff
@@ -247,19 +233,16 @@ export class TxRunner {
 
     let currentDateTime = new Date();
     try {
-      const { transactionHash } = await web3.eth.sendTransaction(
-        tx,
-        undefined,
-        { checkRevertBeforeSending: false, ignoreGasPricing: true }
-      );
-      const receipt = await tryTillReceiptAvailable(transactionHash);
-      tx = await tryTillTxAvailable(transactionHash);
+      const signer = await (provider as BrowserProvider).getSigner();
```

Comment:
`.getSigner();` wll default to get the signer for the account with index 0... not the selected one.

Reply:
updated